### ISSUE TITLE
Fatal regression: Impossible to ungregister from slot with version v3.7.1 (2019080701)

### ISSUE THREAD

**Issue by Kathrin84** [academic-moodle-cooperation/moodle-mod_organizer#73]
Hi guys,

when using your latest published version v3.7.1 (2019080701) on a Moodle 3.7.+ instance it's impossible for a student to unregister from a time slot.

This error is shown:

```
Debug info:
Error code: Inconsistent state: Cannot execute registration action! Please navigate back and refresh your browser!

Stack trace:
line 494 of /lib/setuplib.php: moodle_exception thrown
line 184 of /mod/organizer/view_action.php: call to print_error()
```

You added this code https://github.com/academic-moodle-cooperation/moodle-mod_organizer/blob/MOODLE_37_STABLE/view_action.php#L305, but there's one parameter missing so the variable $ismyslot gets the wrong value and leads to the fact that the return value will always be false in that case.

The solution is this:

```
--- a/mod/organizer/view_action.php
+++ b/mod/organizer/view_action.php
@@ -302,7 +302,7 @@ function organizer_organizer_student_action_allowed($action, $slot) {
 
     list(, , $organizer, $context) = organizer_get_course_module_data();
 
-    list($canregister, $canunregister, $canreregister, $myapp, $regslotx, $myslotexists, , $slotdisabled,
+    list($canregister, $canunregister, $canreregister, $myapp, $regslotx, $myslotexists, , $slotdisabled, ,
             $ismyslot, $slotfull, $disabled, $isalreadyinqueue, $isqueueable)
                 = organizer_get_studentrights($slotx, $organizer, $context);
```

Best, Kathrin

**Comment by goggo24** [comment]
Thanks for reporting. We will fix it asap.

**Comment by abias** [comment]
@goggo24 , this bug is now fixed in 3.7.2 according to the commit history on https://github.com/academic-moodle-cooperation/moodle-mod_organizer/commits/MOODLE_37_STABLE. So this issue could be closed, right?

**Comment by goggo24** [comment]
yes - fixed in v.3.7.2+
